src/hooks/useAuthenticated.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { USER_SELF } from '@/components/constants/url';
+
+export const DEFAULT_AVATAR = '/Avatar.png';
+
+interface Userdata {
```

Comment:
are we planning to use these types as a base and then extend over them?
if not then can we please make them types?

Reply:
No plan of that as of now. Converted it to type